pkgs/os-specific/darwin/apple-source-releases/xnu/10.13.6.nix

```
@@ -0,0 +1,165 @@
+{ appleDerivation', lib, stdenv, stdenvNoCC, buildPackages
+, bootstrap_cmds, bison, flex
+, gnum4, unifdef, perl, python3
+, headersOnly ? true
+}:
+
+appleDerivation' (if headersOnly then stdenvNoCC else stdenv) (
+  let arch = if stdenv.isx86_64 then "x86_64" else "arm64";
+  in
+  {
+  depsBuildBuild = [ buildPackages.stdenv.cc ];
+
+  nativeBuildInputs = [ bootstrap_cmds bison flex gnum4 unifdef perl python3 ];
+
+  patches = lib.optionals stdenv.isx86_64 [
+    ./python3.patch
+    ./0001-Implement-missing-availability-platform.patch
+  ];
+
+  postPatch = ''
+    substituteInPlace Makefile \
+      --replace-fail "/bin/" "" \
+      --replace-fail "MAKEJOBS := " '# MAKEJOBS := '
+
+    substituteInPlace makedefs/MakeInc.cmd \
+      --replace-fail "/usr/bin/" "" \
+      --replace-fail "/bin/" ""
+
+    substituteInPlace makedefs/MakeInc.def \
+      --replace-fail "-c -S -m" "-c -m"
+
+    substituteInPlace makedefs/MakeInc.top \
+      --replace-fail "MEMORY_SIZE := " 'MEMORY_SIZE := 1073741824 # '
+
+    substituteInPlace libkern/kxld/Makefile \
+      --replace-fail "-Werror " ""
+
+    substituteInPlace SETUP/kextsymboltool/Makefile \
+      --replace-fail "-lstdc++" "-lc++ -lc++abi"
```

nit: `-lc++abi` is not needed.
```suggestion
      --replace-fail "-lstdc++" "-lc++"
```
tho i think that clang changes `-lstdc++` to `-lc++` too so the entire substituteInPlace might be unneeded as well as long as the link step is initiated by clang and not ld

---

It still builds without the substitute but I can't find documentation on this. Do you know where to find some?

---

i don't i just stumbled on it. it seems like it is related to the flag `-stdlib=xxx`. when `-stdlib=libstdc++` no translation is made. `-stdlib=libc++`  then `-lstdc++ -> -lc++`. and perhaps the default is `-stdlib=platform` and for darwin seems to be `libc++`.